bazaar/src/tests.rs

```diff
@@ -53,261 +51,213 @@ impl ExtBuilder {
+#[test]
+fn create_new_business_is_ok() {
+    ExtBuilder::build().execute_with(|| {
+        let cid = create_cid();
+
+        assert!(EncointerBazaar::create_business(Origin::signed(alice()), cid, url()).is_ok());
+        assert!(EncointerBazaar::create_business(Origin::signed(bob()), cid, url1()).is_ok());
+
+        assert_eq!(EncointerBazaar::business_registry(cid, alice()), BusinessData::new(url(), 1));
+        assert_eq!(EncointerBazaar::business_registry(cid, bob()), BusinessData::new(url1(), 1));
+    });
+}
+
 #[test]
-fn create_new_shop_works() {
+fn create_business_with_invalid_cid_is_err() {
     ExtBuilder::build().execute_with(|| {
-        // initialisation
-        let cid = register_test_community::<TestRuntime>(None, 2);
-        let alice = AccountId::from(AccountKeyring::Alice);
-        let alice_shop = ShopIdentifier::from("QmW6WLLhUPsosBcKebejveknjrSQjZjq5eYFVBRfugygTB");
-        // upload dummy store to blockchain
-        assert!(
-            EncointerBazaar::new_shop(Origin::signed(alice.clone()), cid, alice_shop.clone())
-                .is_ok()
-        );
-
-        // get shops from blockchain
-        let shops = EncointerBazaar::shop_registry(cid);
-        let alices_shops = EncointerBazaar::shops_owned(cid, alice.clone());
-        // assert that shop was added
-        assert!(shops.contains(&alice_shop));
-        assert!(alices_shops.contains(&alice_shop));
-        // assert that the shop is owned by alice
-        assert_eq!(EncointerBazaar::shop_owner(&cid, &alice_shop), alice);
+        assert!(EncointerBazaar::create_business(Origin::signed(alice()), CommunityIdentifier::zero(), url()).is_err());
+
+        assert_eq!(EncointerBazaar::business_registry(CommunityIdentifier::zero(), alice()), BusinessData::default());
     });
 }
 
 #[test]
-fn create_new_shop_with_bad_cid_fails() {
+fn create_business_duplicate_is_err() {
     ExtBuilder::build().execute_with(|| {
-        // initialisation
-        let alice = AccountId::from(AccountKeyring::Alice);
-        let alice_shop = ShopIdentifier::from("QmW6WLLhUPsosBcKebejveknjrSQjZjq5eYFVBRfugygTB");
-        let cid = CommunityIdentifier::from(blake2_256(&(0, alice.clone()).encode())); // fails to register cid
-
-        // assert that upload fails
-        assert!(
-            EncointerBazaar::new_shop(Origin::signed(alice.clone()), cid, alice_shop.clone())
-                .is_err()
-        );
-
-        // get shops from blockchain
-        let shops = EncointerBazaar::shop_registry(cid);
-        let alices_shops = EncointerBazaar::shops_owned(cid, alice);
-
-        // assert that shop was not added
-        assert_eq!(shops.contains(&alice_shop), false);
-        assert_eq!(alices_shops.contains(&alice_shop), false);
+        let cid = create_cid();
+
+        assert!(EncointerBazaar::create_business(Origin::signed(alice()), cid, url()).is_ok());
+        assert!(EncointerBazaar::create_business(Origin::signed(alice()), cid, url1()).is_err());
```

Comment:
wouldn't it be safer to test for a specific error? Otherwise we're not sure the correct error has been thrown

Reply:
done